jar-infer/jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/BytecodeAnnotator.java

```diff
@@ -279,8 +278,7 @@ public static void annotateBytecodeInJar(
     // Do not use JarInputStream in place of JarFile/JarEntry. JarInputStream misses MANIFEST.MF
     // while iterating over the entries in the stream.
     // Reference: https://bugs.openjdk.java.net/browse/JDK-8215788
-    for (Enumeration<JarEntry> entries = inputJar.entries(); entries.hasMoreElements(); ) {
-      JarEntry jarEntry = entries.nextElement();
+    for (JarEntry jarEntry : inputJar.stream().collect(ImmutableList.toImmutableList())) {
```

Comment:
In these cases can we do a `forEach()` on the stream rather than collecting in a list and looping over that?  That seems a bit cleaner and avoids copying to another list.

Reply:
No, we can't. The code bellow can throw `IOException`. If that happens within a lambda, we need to catch it there, if it happens within an iteration we can propagate it up.

Reply:
Maybe add a code comment to that effect?

Reply:
Also done 😄